Makefile

```diff
@@ -595,53 +608,96 @@ endef
-	$(OPM) validate .konflux/catalog/$(PACKAGE_NAME_KONFLUX)/
-
-.PHONY: konflux-generate-catalog-production ## generate a registry.redhat.io catalog
-konflux-generate-catalog-production: konflux-generate-catalog
-        # overlay the bundle image for production
-	sed -i 's|quay.io/redhat-user-workloads/telco-5g-tenant/$(PACKAGE_NAME_KONFLUX)-bundle-4-21|registry.redhat.io/openshift4/$(PACKAGE_NAME_KONFLUX)-bundle|g' $(CATALOG_KONFLUX)
-        # From now on, all the related images must reference production (registry.redhat.io) exclusively
-	./hack/konflux-validate-related-images-production.sh --set-catalog-file $(CATALOG_KONFLUX)
-	$(OPM) validate .konflux/catalog/$(PACKAGE_NAME_KONFLUX)/
+konflux-validate-catalog: sync-git-submodules opm ## validate the current catalog file
+	$(MAKE) -C $(PROJECT_DIR)/telco5g-konflux/scripts/catalog konflux-validate-catalog \
+		CATALOG_KONFLUX=$(PROJECT_DIR)/$(CATALOG_KONFLUX) \
+		OPM=$(OPM)
+
+.PHONY: konflux-generate-catalog
+konflux-generate-catalog: sync-git-submodules yq opm ## generate a quay.io catalog
+	$(MAKE) -C $(PROJECT_DIR)/telco5g-konflux/scripts/catalog konflux-generate-catalog \
+		CATALOG_TEMPLATE_KONFLUX_INPUT=$(PROJECT_DIR)/$(CATALOG_TEMPLATE_KONFLUX_INPUT) \
+		CATALOG_TEMPLATE_KONFLUX_OUTPUT=$(PROJECT_DIR)/$(CATALOG_TEMPLATE_KONFLUX_OUTPUT) \
+		CATALOG_KONFLUX=$(PROJECT_DIR)/$(CATALOG_KONFLUX) \
+		PACKAGE_NAME_KONFLUX=$(PACKAGE_NAME_KONFLUX) \
+		BUNDLE_BUILDS_FILE=$(PROJECT_DIR)/.konflux/catalog/bundle.builds.in.yaml \
+		OPM=$(OPM) \
+		YQ=$(YQ)
+	$(MAKE) konflux-validate-catalog
+
+.PHONY: konflux-generate-catalog-production
+konflux-generate-catalog-production: sync-git-submodules yq opm ## generate a registry.redhat.io catalog
+	$(MAKE) -C $(PROJECT_DIR)/telco5g-konflux/scripts/catalog konflux-generate-catalog-production \
+		CATALOG_TEMPLATE_KONFLUX_INPUT=$(PROJECT_DIR)/$(CATALOG_TEMPLATE_KONFLUX_INPUT) \
+		CATALOG_TEMPLATE_KONFLUX_OUTPUT=$(PROJECT_DIR)/$(CATALOG_TEMPLATE_KONFLUX_OUTPUT) \
+		CATALOG_KONFLUX=$(PROJECT_DIR)/$(CATALOG_KONFLUX) \
+		PACKAGE_NAME_KONFLUX=$(PACKAGE_NAME_KONFLUX) \
+		BUNDLE_NAME_SUFFIX=$(BUNDLE_NAME_SUFFIX) \
+		PRODUCTION_BUNDLE_NAME=$(PRODUCTION_BUNDLE_NAME) \
+		BUNDLE_BUILDS_FILE=$(PROJECT_DIR)/.konflux/catalog/bundle.builds.in.yaml \
+		OPM=$(OPM) \
+		YQ=$(YQ)
+	$(MAKE) konflux-validate-catalog
+
+.PHONY: konflux-filter-unused-redhat-repos
+konflux-filter-unused-redhat-repos: ## Filter unused repositories from redhat.repo in operator lock folder
+	@echo "Filtering unused repositories from operator runtime lock folder..."
+	$(MAKE) -C $(PROJECT_DIR)/telco5g-konflux/scripts/rpm-lock filter-unused-repos REPO_FILE=$(PROJECT_DIR)/.konflux/operator/redhat.repo
+	@echo "Filtering completed."
+
+.PHONY: konflux-compare-catalog
+konflux-compare-catalog: sync-git-submodules ## Compare generated catalog with upstream FBC image
```

Comment:
did you run the compare catalog target to verify the catalog generated from the template was the same?

Reply:
Both catalogs reference the same bundle SHA256 (ae16e8e5ded756b3d7cbe388429bab4b62b613e9872310e7c69b67edc1db9bab)
All other content (metadata, properties, CRD descriptions, annotations) is identical
The only difference is the image registry/namespace:
Upstream catalog is production (uses registry.redhat.io)
Generated catalog is non-production (uses quay.io)